utils/docker-utils.sh

```diff
@@ -90,34 +90,43 @@ docker::await_service_destroy() {
 docker::service_destroy() {
     local -r SERVICE_NAME=${1:?"FATAL: await_container_destroy SERVICE_NAME not provided"}
 
+    log info "Waiting for service $SERVICE_NAME to be removed ... "
     if [[ -n $(docker service ls -qf name=instant_"${SERVICE_NAME}") ]]; then
-        try "docker service scale instant_${SERVICE_NAME}=0" "Failed to scale down ${SERVICE_NAME}"
-        try "docker service rm instant_${SERVICE_NAME}" "Failed to remove service ${SERVICE_NAME}"
+        try "docker service scale instant_${SERVICE_NAME}=0" catch "Failed to scale down ${SERVICE_NAME}"
+        try "docker service rm instant_${SERVICE_NAME}" catch "Failed to remove service ${SERVICE_NAME}"
         docker::await_service_destroy "${SERVICE_NAME}"
     fi
+    overwrite "Waiting for service $SERVICE_NAME to be removed ... Done"
 }
 
-# Tries to remove a volume and retries until it works with a timeout
+# Tries to remove volumes and retries until it works with a timeout
 #
 # Arguments:
-# - $1 : volume name (eg. es-data)
+# - $1 : volumes names, e.g. "es-data" "psql-1" ..
 #
 docker::try_remove_volume() {
-    local -r VOLUME_NAME=${1:?"FATAL: remove_volume_retry VOLUME_NAME not provided"}
-
-    if ! docker volume ls | grep -q "\sinstant_${VOLUME_NAME}$"; then
-        log info "Tried to remove volume ${VOLUME_NAME} but it doesn't exist on this node"
-        return 1
+    if [[ -z "$*" ]]; then
+        log error "FATAL: try_remove_volume parameter missing"
+        exit 1
     fi
 
-    log info "Waiting for volume ${VOLUME_NAME} to be removed..."
-    local start_time
-    start_time=$(date +%s)
-    until [[ -n "$(docker volume rm instant_"${VOLUME_NAME}" 2>/dev/null)" ]]; do
-        config::timeout_check "${start_time}" "${VOLUME_NAME} to be removed" "60" "10"
-        sleep 1
+    for i in "$@"; do
+        local -r volume_name=${i}
+
+        if ! docker volume ls | grep -q "\sinstant_${volume_name}$"; then
+            log info "Tried to remove volume ${volume_name} but it doesn't exist on this node"
+            return 1
+        fi
```

Comment:
Seems like a good place for a try-throw

Reply:
this will return 1 and not exit, so maybe let's keep it like this ?, try and catch and then return will lead to the same I think

Reply:
On second thought, we probably don't even want it to return... for instance, if we supply a long list of volumes to remove, but the first one returns, then the rest that are possibly on the node won't be removed. Also, maybe a log warn would be better.

Reply:
Yes you're right, fixed in [a0370df](https://github.com/jembi/platform/pull/188/commits/a0370dfd3a72d6c29352623304555257bccbe370)